config.py

```diff
@@ -1,46 +1,105 @@
-import settings
+import os
 
 
 class Config(object):
     # make the warning shut up until Flask-SQLAlchemy v3 comes out
     SQLALCHEMY_TRACK_MODIFICATIONS = True
-    SQLALCHEMY_DATABASE_URI = settings.select_db("Config")
 
-    PYLTI_CONFIG = settings.PYLTI_CONFIG
+    # LTI consumer key and shared secret
+    CONSUMER_KEY = os.environ.get("LTI_KEY")
+    SHARED_SECRET = os.environ.get("LTI_SECRET")
+
+    # Configuration for pylti library. Uses the above key and secret
+    PYLTI_CONFIG = {
+        "consumers": {CONSUMER_KEY: {"secret": SHARED_SECRET}},
+        # Custom configurable roles
+        "roles": {
+            "staff": [
+                "urn:lti:instrole:ims/lis/Administrator",
+                "Instructor",
+                "ContentDeveloper",
+                "urn:lti:role:ims/lis/TeachingAssistant",
+            ]
+        },
+    }
 
     SESSION_COOKIE_NAME = "ft_session"
 
     # Chrome 80 SameSite=None; Secure fix
     SESSION_COOKIE_SECURE = True
     SESSION_COOKIE_SAMESITE = "None"
 
+    SQLALCHEMY_DATABASE_URI = os.environ.get("DATABASE_URI")
+
 
 class BaseConfig(object):
     DEBUG = False
     TESTING = False
 
     # make the warning shut up until Flask-SQLAlchemy v3 comes out
     SQLALCHEMY_TRACK_MODIFICATIONS = True
-    SQLALCHEMY_DATABASE_URI = settings.select_db("BaseConfig")
-
-    PYLTI_CONFIG = settings.PYLTI_CONFIG
 
     SESSION_COOKIE_NAME = "ft_session"
 
     # Chrome 80 SameSite=None; Secure fix
     SESSION_COOKIE_SECURE = True
     SESSION_COOKIE_SAMESITE = "None"
 
+    SQLALCHEMY_DATABASE_URI = os.environ.get("DATABASE_URI")
+
+    # Title of the tool. Appears in the <title> element, headers, and configuration XML
+    TOOL_TITLE = os.environ.get("TOOL_TITLE", "Faculty Tools")
+
+    # Which theme directory to use. Leave blank for default.
+    THEME_DIR = os.environ.get("THEME_DIR", "")
+
+    # Canvas instance URL. ex: https://example.instructure.com/
+    BASE_URL = os.environ.get("BASE_CANVAS_SERVER_URL", "https://example.com")
```

Comment:
Since the BASE_URL is immediately concatenated with `api/v1/`, it's important that it includes a trailing slash. There are other places in the code where a trailing slash is assumed.

1. Ensure the default includes a trailing slash:

```suggestion
    BASE_URL = os.environ.get("BASE_CANVAS_SERVER_URL", "https://example.com/")
```

2. Ensure the above comment and any other documentation referencing BASE_URL states that the trailing slash is required.


Alternatively, consider dropping the trailing slash requirement and altering the code to match. This will require similar documentation stating to NOT include it.

Reply:
fixed.